crates/op-rpc-types/src/op/transaction/mod.rs

```diff
@@ -0,0 +1,188 @@
+use alloy::{
+    consensus::{Signed, TxEip1559, TxEip2930, TxEip4844, TxEip4844Variant, TxEnvelope, TxLegacy},
+    rpc::types::eth::{ConversionError, TransactionRequest},
+};
+use alloy_primitives::{B256, U128, U64};
+use serde::{Deserialize, Serialize};
+
+use self::tx_type::TxType;
+
+pub mod receipt;
+pub mod tx_type;
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
+#[serde(rename_all = "camelCase")]
+pub struct Transaction {
+    /// Ethereum Transaction Types
+    #[serde(flatten)]
+    pub inner: alloy::rpc::types::eth::Transaction,
+    /// The ETH value to mint on L2
+    #[serde(rename = "mint", skip_serializing_if = "Option::is_none")]
+    pub mint: Option<U128>,
```

Comment:
it may be preferable to use Rust's native `u128` type here, that reduces the need for conversions - in line with Alloy's preferences for native types where possible